src/python/pants/util/rwbuf.py

```diff
@@ -34,8 +35,10 @@ def read_from(self, pos, size=-1):
       return self._io.read() if size == -1 else self._io.read(size)
 
   def write(self, s):
+    if not isinstance(s, bytes):
+      raise ValueError('Argument not in bytes: {0}'.format(s))
```

Comment:
Might be more useful to say what type the argument is, e.g., `'Expected bytes, not {}, for argument {}.'.format(type(s), s)`

Side note - we typically don't enumerate the format args, unless we're repeating them (in which case we prefer to give them names instead of ordinals).